published/external/xdpapi.h

```diff
@@ -40,15 +49,84 @@ XDP_CREATE_PROGRAM_FN(
     _Out_ HANDLE *Program
     );
 
+#else
+
 typedef
-HRESULT
+XDP_STATUS
+XDP_CREATE_PROGRAM_FN(
+    _In_ UINT32 InterfaceIndex,
+    _In_ CONST XDP_HOOK_ID *HookId,
+    _In_ UINT32 QueueId,
+    _In_ XDP_CREATE_PROGRAM_FLAGS Flags,
+    _In_reads_(RuleCount) CONST XDP_RULE *Rules,
+    _In_ UINT32 RuleCount,
+    _Out_ HANDLE *Program
+    );
+
+#endif
+
+typedef
+XDP_STATUS
 XDP_INTERFACE_OPEN_FN(
     _In_ UINT32 InterfaceIndex,
     _Out_ HANDLE *InterfaceHandle
     );
 
 #include "afxdp.h"
 
+typedef
+VOID *
+XDP_GET_ROUTINE_FN(
+    _In_z_ const CHAR *RoutineName
+    );
+
+#if defined(_KERNEL_MODE)
+
+typedef
+VOID
+XDP_DELETE_PROGRAM_FN(
+    _In_ HANDLE Program
+    );
+
+typedef
+VOID
+XDP_INTERFACE_CLOSE_FN(
+    _In_ HANDLE InterfaceHandle
+    );
+
+DEFINE_GUID(
+    NPI_XDPAPI_INTERFACE_ID,
+    0x1683b7b0, 0xcf44, 0x4757, 0x91, 0xf6, 0xc6, 0x19, 0x9b, 0x9d, 0x00, 0xfe);
+
+//
+// The only API version currently supported. Any change to the API is considered
```

Comment:
Why is "any change" considered breaking?

Reply:
oh, woops, this comment should have been removed with 1.0 release.

Reply:
Addressed with #454